perl/bin/cgpVaf.pl

```diff
@@ -40,33 +40,28 @@ BEGIN
 use Capture::Tiny qw(:all);
 
 use Log::Log4perl;
+use Log::Log4perl::Level;
 Log::Log4perl->init("$Bin/../config/log4perl.vaf.conf");
 use lib "$Bin/../lib";
 use Sanger::CGP::Vaf::Data::ReadVcf;
 use Sanger::CGP::Vaf::VafConstants;
 
 my $log = Log::Log4perl->get_logger(__PACKAGE__);
 
-my $store_results;
-
-my $debug = 0;
-
 my $tags=$Sanger::CGP::Vaf::VafConstants::SNP_TAGS;
 
 try {
     my ($options) = option_builder();
-    if ($options->{'dbg'}){
-        $log->debug("================Using Parameters===========================");
-      $log->debug(Dumper($options));
-    }
+    $log->debug("================Using Parameters===========================");
+    $log->debug(Dumper($options));
     if ($options->{'a'} eq 'indel') {
         $tags=$Sanger::CGP::Vaf::VafConstants::INDEL_TAGS;
     }
     my $vcf_obj = Sanger::CGP::Vaf::Data::ReadVcf->new($options);
-    
-    my $progress_hash;
-    # progress checked before the processing starts , speed ups concatenation step 
-    my ($chromosomes)=$vcf_obj->getChromosomes($options->{'chr'});
+
+    # progress checked before the processing starts , speed ups concatenation step
+    my ($chromosomes, $chr_count)=$vcf_obj->getChromosomes($options->{'chr'});
+    $log->logcroak('It appears that no data has been detected for processing, please check command line params.') if($chr_count == 0);
```

Comment:
Assumes chr_count is always present if vcf/bed is provided, would it run for user provided chr list and then merge and concatenate the same.

Reply:
based on getChromosomes I can see it will(?)

Reply:
wait if user chooses concatenate option then $chromosomes reverts to all in the vcf files, so it will not concatenate user only chromosomes ( relevant only if we want this option ?)

Reply:
My understanding, based on the documentation in the wiki, is that you should run with `-chr` and then run `-ct` as a separate step (without `-chr`).

`-ct` concatenates files that exist, it doesn't check anything.  Regardless of this it functions.

Reply:
@kr2 when `-ct` is ON,  chromosomes will be populated from vcf file and it will run all the chromosomes again before going to concatenation step so chr==0 will not be true, seems unnecessary to me

Reply:
see debug on Line 234:
  $log->debug('Concatenation option selected, chromosome option will be set to all chromosomes')

Reply:
@keiranmraine as you said may be let this release go as it is now and fix this ( if required in later release).

Reply:
Why would it rerun everything?  I thought that was the point of, this:

https://github.com/cancerit/vafCorrect/blob/393301a903a41b86ac34ff34510ad638f78625df/perl/bin/cgpVaf.pl#L65

calling:

https://github.com/cancerit/vafCorrect/blob/393301a903a41b86ac34ff34510ad638f78625df/perl/lib/Sanger/CGP/Vaf/Data/ReadVcf.pm#L147-L163

Reply:
The concatenation step is reading from `$options->{'chr'}`.  This is not modified by anything.  The chr lists used for processing are distinct memory references.

Reply:
$chromosomes is populated to all chromosomes in vcf  if user_chr is empty since $options->{'chr'} is set empty when `-ct` is ON.

Reply:
With `-ct`:

```
kr2@kr2-develop:~$ sudo rm -rf result; mkdir -p result
kr2@kr2-develop:~$ docker run -ti --rm -v $HOME:/data `docker images -q | head -n 1` bash -c 'perl -d `which cgpVaf.pl` -m -r 1 \
-hdr /data/ref/shared/ucscHiDepth_0.01_mrg1000_no_exon_coreChrs.bed.gz \
-e .vcf.gz \
-oe .vcf \
-d /data/6855_amplicon \
-o /data/result \
-a indel \
-g /data/ref/genome.fa \
-pid 1 \
-tn PD26840a \
-nn PD26840b \
-vcf /data/6855_amplicon/mini.pindel.flagged.vcf.gz \
-ct 1'

Loading DB routines from perl5db.pl version 1.49
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.
...
44:	Log::Log4perl->init("$Bin/../config/log4perl.vaf.conf");
  DB<1> c 107
[INFO] 2019/10/22 15:08:47 main(214): Analysing: indel

 >>>>>> To view overall progress log please check vcfcommons.log file created in the current directory >>>>>>>>>

 >>>>>> Samples specific <CHR>_progress.out file is created in the output directory : /data/result >>>>>>>>>
Analysing 3
[INFO] 2019/10/22 15:09:15 Sanger::CGP::Vaf::Data::ReadVcf(771): Completed analysis for chromosome: 3 
main::CODE(0x34f3c38)(/opt/wtsi-cgp/bin/cgpVaf.pl:107):
107:	    if($options->{'ct'} || @{$options->{'chr'}} == 0 ) {
  DB<2> print Dumper($options->{'chr'});
$VAR1 = [];
```

Without `-ct`:

```
kr2@kr2-develop:~$ sudo rm -rf result; mkdir -p result
kr2@kr2-develop:~$ docker run -ti --rm -v $HOME:/data `docker images -q | head -n 1` bash -c 'perl -d `which cgpVaf.pl` -m -r 1 \
-hdr /data/ref/shared/ucscHiDepth_0.01_mrg1000_no_exon_coreChrs.bed.gz \
-e .vcf.gz \
-oe .vcf \
-d /data/6855_amplicon \
-o /data/result \
-a indel \
-g /data/ref/genome.fa \
-pid 1 \
-tn PD26840a \
-nn PD26840b \
-vcf /data/6855_amplicon/mini.pindel.flagged.vcf.gz'

Loading DB routines from perl5db.pl version 1.49
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.
...
44:	Log::Log4perl->init("$Bin/../config/log4perl.vaf.conf");
  DB<1> c 107
[INFO] 2019/10/22 15:10:48 main(214): Analysing: indel

 >>>>>> To view overall progress log please check vcfcommons.log file created in the current directory >>>>>>>>>

 >>>>>> Samples specific <CHR>_progress.out file is created in the output directory : /data/result >>>>>>>>>
Analysing 3
[INFO] 2019/10/22 15:11:17 Sanger::CGP::Vaf::Data::ReadVcf(771): Completed analysis for chromosome: 3 
main::CODE(0x336c918)(/opt/wtsi-cgp/bin/cgpVaf.pl:107):
107:	    if($options->{'ct'} || @{$options->{'chr'}} == 0 ) {
  DB<2> print Dumper($options->{'chr'});
$VAR1 = [];
```

Reply:
yes it will be set empty when `-ct` is true which then populates $chromosomes to all in vcf  file.
if you print Dumper($chromosomes) it will be populated from vcf

Reply:
Yes I'm aware that `$chromosomes` is populated from bed or vcf, but the progress check will skip them regardless. I'm not sure if we're agreeing that this works or disagreeing.